hasjob/models/campaign.py

```diff
@@ -164,6 +164,21 @@ class Campaign(BaseNameMixin, db.Model):
         lambda obj: obj.start_at <= datetime.utcnow() < obj.end_at,
         lambda cls: db.and_(cls.start_at <= db.func.utcnow(), cls.end_at > db.func.utcnow()),
         label=__("Live"))
+    state.add_conditional_state('CURRENT', state.ENABLED,
+        lambda obj: obj.start_at <= obj.start_at <= datetime.utcnow() < obj.end_at <= datetime.utcnow() + timedelta(days=30),
+        lambda cls: db.and_(
+            cls.start_at <= db.func.utcnow(),
+            cls.end_at > db.func.utcnow(),
+            cls.end_at <= datetime.utcnow() + timedelta(days=30)),
+        label=__("Current"))
+    state.add_conditional_state('LONGTERM', state.ENABLED,
+        lambda obj: obj.start_at <= obj.start_at <= datetime.utcnow() < datetime.utcnow() + timedelta(days=30) < obj.end_at,
+        lambda cls: db.and_(cls.start_at <= datetime.utcnow(), cls.end_at > datetime.utcnow() + timedelta(days=30)),
+        label=__("Long term"))
+    state.add_conditional_state('OFFLINE', state.ENABLED,
+        lambda obj: obj.start_at > datetime.utcnow() or obj.end_at <= datetime.utcnow(),
+        lambda cls: db.or_(cls.start_at > db.func.utcnow(), cls.end_at <= db.func.utcnow()),
+        label=__("Offline"))
```

Comment:
I've been using tuples for labels as I assume almost everywhere that label is a NamedTuple and it'll have name and title. Should we do that here as well?

Reply:
Ideally NameTuple everywhere because we shouldn’t expose integer primary keys to the outside world, even in an API. The `name` portion is then a secondary key. In this case, it doesn’t matter because the state is a boolean, but we could add a name anyway for consistency.